server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -353,6 +354,7 @@ public DesiredBalance compute(
                     TimeValue.timeValueMillis(currentTime - computationStartedTime).toString(),
                     i
                 );
+                finishReason = DesiredBalance.ComputationFinishReason.YIELD_TO_NEW_INPUT;
```

Comment:
I'm wondering about the scenario where we would want to return `DesiredBalance.ComputationFinishReason.STOP_EARLY` to get some primary shards allocated ASAP (`assignedNewlyCreatedPrimaryShards` is true), but we do this `isFresh` check first and discard the result in `processInput` 🤔  Should we do the other check first? Thoughts?

Reply:
hmm, maybe. IIUC, the case you're thinking of would be if we keep constantly getting interrupted by a new reroute request where each computation takes a bit time, not long enough to hit the new 1s threshold, but hight enough that when accumulated across consecutive allocate calls it leads to a large delay of assigning shards. I am not sure how realistic of an issue that can be. I would refrain from further complicating this unless there is concrete evidence of an issue.